node/rest/src/routes.rs

```diff
@@ -936,4 +936,21 @@ impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
             None => Err(RestError::service_unavailable(anyhow!("Route isn't available for this node type"))),
         }
     }
+
+    /// GET /{network}/record_count
+    pub(crate) async fn get_number_of_records(State(rest): State<Self>) -> ErasedJson {
```

Comment:
This endpoint will be quite expensive, can we put it behind auth to prevent DoSing nodes? - https://github.com/ProvableHQ/snarkVM/pull/3047#discussion_r2590633826

Reply:
as mentioned in the linked thread, this operation is in general less expensive than just obtaining a block; what should the criteria for auth-gating be? some fixed amount of time on reference hardware, or just algorithmic complexity (even if it's unlikely to be a concern until very large numbers are reached)?